azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceContextLiveTest.java

```diff
@@ -0,0 +1,292 @@
+      Set<? extends NodeMetadata> nodes = view.getComputeService().suspendNodesMatching(inGroup(groupName));
+      assertTrue(nodes.size() > 0);
+
+      boolean allStopped = false;
+      while (!allStopped) {
+         nodes = view.getComputeService().listNodesDetailsMatching(nameStartsWith(groupName));
+         for (NodeMetadata node : nodes) {
+            if (node.getStatus() != NodeMetadata.Status.SUSPENDED)
+            {
+               // Not stopped yet
+               allStopped = false;
+               try {
+                  Thread.sleep(15 * 1000);
+               } catch (InterruptedException e) {
+               }
+               continue;
+            }
+            else
+            {
+               allStopped = true;
+            }
+         }
+      }
+      assertTrue(allStopped);
+   }
+
+   @Test(dependsOnMethods = "testStop")
+   public void testStart() throws RunScriptOnNodesException {
+      final String groupName = String.format(NAME_PREFIX, System.getProperty("user.name").substring(0, 3));
+      Set<? extends NodeMetadata> nodes = view.getComputeService().resumeNodesMatching(inGroup(groupName));
+      assertTrue(nodes.size() > 0);
+
+      boolean allStarted = false;
+      while (!allStarted) {
+         nodes = view.getComputeService().listNodesDetailsMatching(nameStartsWith(groupName));
+         for (NodeMetadata node : nodes) {
+            if (node.getStatus() != NodeMetadata.Status.RUNNING)
+            {
+               // Not started yet
+               allStarted = false;
+               try {
+                  Thread.sleep(15 * 1000);
+               } catch (InterruptedException e) {
+               }
+               continue;
+            }
+            else
+            {
+               allStarted = true;
+            }
+         }
+      }
+      assertTrue(allStarted);
+   }
+
+   @Test(dependsOnMethods = "testStart")
+   public void testRestart() throws RunScriptOnNodesException {
+      final String groupName = String.format(NAME_PREFIX, System.getProperty("user.name").substring(0, 3));
+      Set<? extends NodeMetadata> nodes = view.getComputeService().rebootNodesMatching(inGroup(groupName));
+      assertTrue(nodes.size() > 0);
+
+      boolean allRestarted = false;
+      while (!allRestarted) {
+         nodes = view.getComputeService().listNodesDetailsMatching(nameStartsWith(groupName));
+         for (NodeMetadata node : nodes) {
+            if (node.getStatus() != NodeMetadata.Status.RUNNING)
+            {
+               // Not started yet
+               allRestarted = false;
+               try {
+                  Thread.sleep(30 * 1000);
+               } catch (InterruptedException e) {
+               }
+               continue;
+            }
+            else
+            {
+               allRestarted = true;
+            }
+         }
+      }
+      assertTrue(allRestarted);
+
+      view.getComputeService().destroyNodesMatching(inGroup(groupName));
+   }
```

Comment:
Also a subclass of the `BaseTemplateBuilderLiveTest` must be added. As usual, you can take the [DigitalOcean](https://github.com/jclouds/jclouds/blob/master/providers/digitalocean2/src/test/java/org/jclouds/digitalocean2/compute/DigitalOcean2TemplateBuilderLiveTest.java) one as an example. Providers should have those tests passing, and try to avoid overriding the base class tests (as much as possible) as that is the contract to guarantee the provider is aligned with the jclouds interfaces and really portable.